components/pages/header.vue

```diff
@@ -1,13 +1,297 @@
+  },
+  beforeUnmount () {
+    window.removeEventListener('scroll', this.handleHeaderScroll)
+  },
+  methods: {
+    gotoPage (href) {
+      this.$router.push(href)
+    },
+    handleHeaderScroll () {
+      const header = document.querySelector('header')
+      if (window.scrollY > 60) {
+        header.classList.add('scroll')
+      } else {
+        header.classList.remove('scroll')
+      }
+    },
+    toggleMobileMenu () {
+      this.isMobileMenuActive = !this.isMobileMenuActive
+    }
+  }
+
 }
 </script>
 
 <style lang="scss" scoped>
+  .header {
+    position: fixed;
+    z-index: 100;
+    width: 100%;
+    height: 156px;
+    transition: all 300ms linear;
+
+    &.scroll {
+      background-color: rgba(37, 43, 66, 0.95);
+      box-shadow: 0 16px 14px -8px rgba(0, 0, 0, 0.3);
+    }
+
+    &__nav {
+      max-width: 1210px;
+      height: 100%;
+      margin: 0 auto;
+      display: flex;
+      justify-content: space-between;
+    }
+
+    &__left, &__socials {
+      display: flex;
+      align-items: center;
+    }
+
+    &__menu {
+      display: flex;
+      margin-left: 40px;
+      gap: 21px;
+
+      &__item a {
+        font-weight: 400;
+        font-size: 1.5rem;
+        line-height: 28px;
+        text-align: center;
+        letter-spacing: 0.2px;
+        color: #FFFFFF;
+        text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+
+        &:hover {
+          color: var(--primary-color);
+        }
+      }
+    }
+
+    &__logo {
+      width: 186px;
+      height: 58px;
+      margin-left: 25px;
+    }
+
+    &__socials {
+      height: 100%;
+      margin-right: 40px;
+      gap: 24px;
+
+      &__item:hover {
+        background-color: var(--primary-color);
+      }
+    }
+  }
+
+  .mobile-nav, .mobile-menu {
+    display: none;
+  }
+
+  @media (max-width: 1110px) {
+    .header {
+      height: 112px;
+    }
+  }
+
+  @media (max-width: 820px) {
+    .header {
+      height: 112px;
+
+      &__logo {
+        display: none;
+      }
+
+      &__menu__item a {
+        font-size: 1.4rem;
+      }
+
+      &__socials__item img {
+        width: 30px;
+        height: 100%;
+      }
+    }
+  }
+  @media (max-width: 650px) {
+    .header {
+      width: 100%;
+      height: 112px;
+
+      &__nav {
+      display: none;
+    }
+    }
+
+    .mobile-nav {
+      display: flex;
+      align-items: center;
+      justify-content: space-between;
+      padding: 17px 16px;
+
+      &__icon {
+        cursor: pointer;
+      }
+    }
+
+    .mobile-menu {
+      position: fixed;
+      height: 100vh;
+      width: 100%;
+      top: 0;
+      left: 100%;
+      background-color: #333;
+      display: flex;
+      justify-content: center;
+      padding-top: 100px;
+      transition: all 400ms ease;
+
+      &.active {
+        left: 0;
+      }
+
+      &.active .close-icon {
+        opacity: 1;
+        pointer-events: auto;
+        cursor: pointer;
+      }
+
+      & .close-icon {
```

Comment:
```suggestion
      .close-icon {
```

`&` is not necessary